include/revng/ABI/CallingConventionTrait.h

```diff
@@ -0,0 +1,1083 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include "revng/Model/Binary.h"
+#include "revng/Model/Register.h"
+#include "revng/Model/Type.h"
+
+namespace abi {
+
+/// A trait struct used for describing the specifics of the calling convention
+/// for an ABI relevant for revng model.
+///
+/// To register a new ABI for use with the `abi::ABI` template, add
+/// a specialization of this trait.
```

Comment:
We're still of the idea that (in the future) this will become a run-time data structure, right?
Put a note about this.

Reply:
Yep, I'm a bit afraid of the vector register side of things. But should be doable.

Reply:
[Done](https://github.com/revng/revng/pull/98/commits/52032ea51413f0900235a9d5274167dd9e60de5f).